src/WebJobs.Extensions.Http/AuthenticatedUserBindingProvider.cs

```diff
@@ -0,0 +1,168 @@
+            }
+
+            private static Task<IValueProvider> BindAsync(IReadOnlyDictionary<string, object> bindingData)
+            {
+                return Task.FromResult<IValueProvider>(new AuthenticatedUserValueProvider(bindingData));
+            }
+
+            public ParameterDescriptor ToParameterDescriptor()
+            {
+                return new ParameterDescriptor
+                {
+                    Name = _parameter.Name,
+                    DisplayHints = new ParameterDisplayHints
+                    {
+                        Description = "AuthenticatedUser"
+                    }
+                };
+            }
+
+            private class AuthenticatedUserValueProvider : IValueProvider
+            {
+                private static HttpClient _client = new HttpClient();
+                private IReadOnlyDictionary<string, object> _bindingData;
+
+                public AuthenticatedUserValueProvider(IReadOnlyDictionary<string, object> bindingData)
+                {
+                    _bindingData = bindingData;
+                }
+
+                public Type Type => typeof(AuthenticatedUser);
+
+                public async Task<object> GetValueAsync()
+                {
+                    var initialRequest = _bindingData.Values.FirstOrDefault(val => val.GetType() == typeof(HttpRequestMessage)) as HttpRequestMessage;
+                    if (initialRequest == null)
+                    {
+                        throw new InvalidOperationException("Cannot determine authenticated user without HttpTrigger.");
+                    }
+
+                    var authorizationHeader = GetAuthorizationHeaderValue(initialRequest);
+                    if (authorizationHeader != null)
+                    {
+                        return await GetAuthenticatedUserFromEasyAuth(authorizationHeader);
+                    }
+                    else
+                    {
+                        return GetAuthenticatedUserFromFunctionKey();
+                    }   
+                }
+
+                private static string GetAuthorizationHeaderValue(HttpRequestMessage initialRequest)
+                {
+                    var idTokenHeaders = initialRequest.Headers.Where(header => header.Key.EndsWith("id-token", StringComparison.OrdinalIgnoreCase));
+                    if (!idTokenHeaders.Any())
+                    {
+                        return null;
+                    }
+                    else
+                    {
+                        var idTokenHeader = idTokenHeaders.First();
+                        var idTokenValue = idTokenHeader.Value.First();
+                        return "Bearer " + idTokenValue;
+                    }
+                }
+
+                private async Task<AuthenticatedUser> GetAuthenticatedUserFromEasyAuth(string authorizationHeader)
+                {
+                    string hostname = Environment.GetEnvironmentVariable("WEBSITE_HOSTNAME");
```

Comment:
@cgillum and I had discussed in my original PR a way to have EA flow the claims as headers to us so we don't have to make an additional loop back request.